src/main/java/com/sprint/mission/discodeit/security/jwt/JwtAuthenticationFilter.java

```diff
@@ -0,0 +1,113 @@
+            new AntPathRequestMatcher("/api/auth/logout", "POST"),
+            new AntPathRequestMatcher("/api/binaryContents/**", "GET"),
+            new AntPathRequestMatcher("/api/auth/csrf-token", "GET"),
+            new AntPathRequestMatcher("/"),
+            new AntPathRequestMatcher("/index.html"),
+            new AntPathRequestMatcher("/swagger-ui/**"),
+            new AntPathRequestMatcher("/v3/api-docs/**"),
+            new AntPathRequestMatcher("/actuator/**"),
+            new AntPathRequestMatcher("/favicon.ico"),
+            new AntPathRequestMatcher("/assets/index-*.js"),
+            new AntPathRequestMatcher("/assets/index-*.css"),
+            new AntPathRequestMatcher("/static/**")
+    );
+
+    @Override
+    protected void doFilterInternal(HttpServletRequest request, HttpServletResponse response,
+            FilterChain filterChain) throws ServletException, IOException {
+
+        if (!requiresAuthentication(request)) {
+            filterChain.doFilter(request, response);
+            return;
+        }
+
+        String authHeader = request.getHeader("Authorization");
+
+        if (authHeader == null || !authHeader.startsWith("Bearer ")) {
+            log.warn("Authorization header is missing");
+            response.sendError(HttpServletResponse.SC_UNAUTHORIZED);
+            return;
+        }
+
+        String token = authHeader.substring(7);
+
+        if (!jwtService.validateToken(token)) {
+            log.warn("Invalid token");
+            response.sendError(HttpServletResponse.SC_UNAUTHORIZED);
+            return;
+        }
+
+        if (!jwtSessionRepository.existsByAccessToken(token)) {
+            log.warn("DB not found access token");
+            response.sendError(HttpServletResponse.SC_UNAUTHORIZED);
+            return;
+        }
+
+        try {
+            Claims claims = jwtService.getClaims(token);
+            Map<String, Object> userDtoMap = claims.get("userDto", Map.class);
+            UserDto userDto = objectMapper.convertValue(userDtoMap, UserDto.class);
+
+            CustomUserDetails userDetails =
+                    (CustomUserDetails) userDetailsService.loadUserByUsername(
+                            userDto.getUsername());
+
+            UsernamePasswordAuthenticationToken authenticationToken =
+                    new UsernamePasswordAuthenticationToken(userDetails, null,
+                            userDetails.getAuthorities());
+
+            SecurityContextHolder.getContext().setAuthentication(authenticationToken);
+            filterChain.doFilter(request, response);
+            log.debug("Authentication principal class: {}",
+                    authenticationToken.getPrincipal().getClass().getName());
+
+        } catch (Exception e) {
+            log.error("토큰 파싱 오류", e);
+            response.sendError(HttpServletResponse.SC_UNAUTHORIZED);
+        }
+    }
+
+    private boolean requiresAuthentication(HttpServletRequest request) {
```

Comment:
인증 자체를 요청하는게 아니라 인증이 필요한지 판단하는 메서드이기에 메서드명을 좀더 책임에 맞게 바꿔봐도 좋을 것 같아요.  
```suggestion
    private boolean isAuthenticationRequired(HttpServletRequest request) {
```